File: frontend/angular-book-ecommerce/src/app/common/order-dto.ts
import { CartItemDTO } from "./cart-item-dto";

export class OrderDTO {
    order_id: number;
  user_id: number;
  total_price: number;
  order_date: Date;
  status: string;
  cart_items: CartItemDTO[];
  shipping_method: string;
  shipping_address: string;
  shipping_date: Date;
  payment_method: string;
  coupon_code: string;

  constructor(data: any) {
    this.order_id = data.order_id;
    this.user_id = data.user_id;
    this.total_price = data.total_price;
    this.order_date = data.order_date;
    this.status = data.status;
    this.cart_items = data.cart_items;
    this.shipping_method = data.shipping_method;
    this.shipping_address = data.shipping_address;
    this.shipping_date = data.shipping_date;
    this.payment_method = data.payment_method;
    this.coupon_code = data.coupon_code;
  }
}

File: backend/spring-boot-book-ecommerce/src/main/java/com/luvannie/springbootbookecommerce/dto/OrderDTO.java
package com.luvannie.springbootbookecommerce.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderDTO {
    @JsonProperty("order_id")
    private Long orderId;

    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("total_price")
    private Double totalPrice;

    @JsonProperty("dateCreated")
    private Date dateCreated;

    @JsonProperty("status")
    private String status;

    @JsonProperty("cart_items")
    private List<CartItemDTO> cartItems;

    @JsonProperty("shipping_method")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    private String shippingAddress;

//    @JsonProperty("shipping_date")
//    private LocalDate shippingDate;

    @JsonProperty("payment_method")
    private String paymentMethod;

    @JsonProperty("coupon_code")
    private String couponCode;



}